crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -313,13 +349,11 @@ where
                 let () = self.size_threshold.validate(message)?;
                 self.try_convert_measurement(message)
             }
-            topic if topic.name.starts_with("tedge/alarms") => {
-                let () = self.size_threshold.validate(message)?;
-                self.alarm_converter.try_convert_alarm(message)
-            }
-            topic if topic.name.starts_with(INTERNAL_ALARMS_TOPIC) => {
-                self.alarm_converter.process_internal_alarm(message);
-                Ok(vec![])
+            topic
+                if topic.name.starts_with("tedge/alarms")
+                    | topic.name.starts_with(INTERNAL_ALARMS_TOPIC) =>
+            {
+                Ok(self.process_alarm_messages(topic, message)?)
```

Comment:
```suggestion
                self.process_alarm_messages(topic, message)
```

It seems that this function does return the same type as `process_alarm_messages()`, so I guess this should work.

Reply:
Yes, it works.